dlpy/model.py

```diff
@@ -2012,83 +2029,78 @@ def display(self, layer_id, filter_id=None):
         plt.show()
 
 
-class DLPyDict(collections.MutableMapping):
-    """ Dictionary that applies an arbitrary key-altering function before accessing the keys """
-
-    def __init__(self, *args, **kwargs):
-        for k in kwargs:
-            self.__setitem__(k, kwargs[k])
-
-    def __getitem__(self, key):
-        return self.__dict__[self.__keytransform__(key)]
-
-    def __setitem__(self, key, value):
-        if value is not None:
-            self.__dict__[self.__keytransform__(key)] = value
-        else:
-            if key in self.__dict__:
-                self.__delitem__[key]
-
-    def __delitem__(self, key):
-        del self.__dict__[self.__keytransform__(key)]
-
-    def __iter__(self):
-        return iter(self.__dict__)
-
-    def __len__(self):
-        return len(self.__dict__)
-
-    def __keytransform__(self, key):
-        return key.lower().replace("_", "")
-
-    def __str__(self):
-        return str(self.__dict__)
-
-
-class Solver(DLPyDict):
+class _Solver(DLPyDict):
```

Comment:
I think we should do something like this _Solver expands Solver so that we don't really remove the Solver and it will be backward compatible.

Reply:
The reason I want to make it an abstract class is that Solver is not used to be instantiated. In its child class, we call set_method to add the property. Although we can directly create a Solver object since the method default as Vanilla, Solver will be exactly identical to VanilaSolver. In coding design perspective, I don't think we should keep Solver and instantiate Solver is not right.

Reply:
we cannot remove Solver, any notebook so far would not work with this removal. Please find another way.